src/MediatR/Internal/RequestHandlerWrapper.cs

```diff
@@ -64,7 +66,11 @@ internal class RequestHandlerWrapperImpl<TRequest, TResponse> : RequestHandlerWr
             return serviceFactory
                 .GetInstances<IPipelineBehavior<TRequest, TResponse>>()
                 .Reverse()
-                .Aggregate((RequestHandlerDelegate<TResponse>) Handler, (next, pipeline) => () => pipeline.Handle((TRequest)request, cancellationToken, next))();
+                .OrderBy(i => i.GetType()
```

Comment:
How do we keep the order consistent, do we care so much? Just wondering if something has the same priority as something else, it could keep jumping around in execution order - this may not be a problem because the priorities are the same - but just thinking about it being a possible nuance that people rely on too heavily thinking they get a guaranteed order of execution.

Reply:
If we have middleware the order of execution does matter a lot. They can enrich the request or context with information the lower order middleware depend on.

The `OrderBy` function is an implementation of [stable sort](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.orderby?redirectedfrom=MSDN&view=netcore-3.1#System_Linq_Enumerable_OrderBy__2_System_Collections_Generic_IEnumerable___0__System_Func___0___1__). So it won't jump around for the things that are of the same order. 

This implementation doesn't get in the way of anyone who doesn't want to use it but helps a developer guarantee the high/low priority order of something like a logging or exception handling middleware.